src/boundary_condition_routines.f90

```diff
@@ -754,6 +754,9 @@ SUBROUTINE BOUNDARY_CONDITIONS_FINALISE(BOUNDARY_CONDITIONS,ERR,ERROR,*)
                   & " is not associated",ERR,ERROR,*999)
           ENDIF
         ENDDO !variable_idx
+        NULLIFY(BOUNDARY_CONDITIONS%SOLVER_EQUATIONS%SOLVER%SOLVER_EQUATIONS)
```

Comment:
I'm not sure why you are nullifying the solver solver equations here? This will result in a memory leak as the solver equations memory will not be freed with the solver is finalised???